SMDS_3/include/CGAL/Mesh_complex_3_in_triangulation_3.h

```diff
@@ -1205,11 +1205,90 @@ class Mesh_complex_3_in_triangulation_3
   std::istream& operator>>(std::istream& is,
     Mesh_complex_3_in_triangulation_3<Tr2, CoI, CuI>& c3t3);
 
+  template <typename Tr2, typename CoI, typename CuI>
+  friend
+  std::ostream & operator<<(std::ostream& os,
+    const Mesh_complex_3_in_triangulation_3<Tr2, CoI, CuI> &c3t3);
+
+
   static std::string io_signature()
   {
     return Get_io_signature<Tr>()();
   }
 
+  std::istream& read_edges(std::istream& is, const std::vector< Vertex_handle >& vertices_handles)
+  {
+    // Reads 1D features
+    std::size_t nEdges;
+    if(IO::is_ascii(is))
+    {
+      is >> nEdges;
+    }
+    else
+    {
+      read(is, nEdges);
+    }
+    // If the file did not have edges
+    if (!is) {
+      return is;
+    }
+
```

Comment:
We should be compatible with the "old" binary files, without the edges.

So, here, if the state of `is` is `eofbit|failbit`, then the `failbit` should be cleared, so that the `file_input` succeeds.